api/lib/infrastructure/http/errors-helper.js

```diff
@@ -0,0 +1,22 @@
+function getErrorDetails(response, customMessage) {
+  let errorDetails;
+  if (response.data.lenght > 0) {
+    errorDetails = JSON.stringify(response.data);
+  } else {
+    errorDetails =
+      response.data.length === 0
+        ? 'Pas de détails disponibles'
+        : { errorDescription: response.data.error_description, errorType: response.data.error };
+  }
```

Comment:
Je n'ai pas bien compris le but de cet condition.

Ce que je veux dire

- si `data` a des données on va utiliser `JSON.stringify(..)`
- sinon
  - si aucune donnée on fourni une chaîne de caractères
  - sinon (ce qui veut dire qu'il y a des données, donc équivalent à la 1ère condition)
    - on retourne un objet

Reply:
C'est du a http-agent, ça fait ça quand y'a une erreur lors de l'appel api
  ```
  let code = null;
      let data;

      if (httpErr.response) {
        code = httpErr.response.status;
        data = httpErr.response.data;
      } else {
        data = httpErr.message;
      }
```
J'ai vu que **response.data** pouvait être '', 'Error' ou `{
        error: 'invalid_client',
        error_description: 'Invalid authentication method for accessing this endpoint.',
      },` mais y'a peut-être d'autres formats de réponse
Du coup, ce qu'il y a dans  **response.data** , comme j'en suis pas toujours très sure alors j'ai essayé de traiter ces 3 cas.

En vrai y'a aussi un cas où on a un object et ça risque de s'afficher comme ça si on fait pas un `JSON.stringify(response.data);` voir datadog ci-dessous

![error object object](https://user-images.githubusercontent.com/7688741/196355071-242ad62c-c5b0-4faa-a741-46a52c04b398.png)

Je rajoute des cas de tests sur `api/tests/unit/infrastructure/http/errors-helper_test.js` pour voir comment améliorer ça

Reply:
En faite ce que je voulais pointer c'est que la ligne 4 et la ligne 9 sont des expressions qui touchent à la même condition `response.data.length > 0`.

Dans ce cas est-ce qu'il n'est pas possible d'utiliser le header Content-Type de la réponse qui permet de connaître le type de la réponse ?

Reply:
Refacto effectué